### PR TITLE
[TBDGen][Test] Check implied Obj-C symbol generation based on a stable swift version

### PR DESCRIPTION
**Pull request by cyndyishida**
I previously thought the coupled `OBJC_CLASS | OBJC_METACLASS` was specific to iOS, but it actually seems tied to compiling for deployment versions lower than when swift was shipped with darwin OSes. The test has been updated to use a recent-ish target triple across all CI configurations, where both symbols get emitted.

### CODE REVIEW

**Comment by cyndyishida** [comment]
@swift-ci please test

File: test/TBD/multimodule-implied-objc.swift [hunk 1]
```diff
@@ -1,32 +1,35 @@
...
 // RUN:   -module-name IndirectMixedDependency -I %t \
-// RUN:   -enable-library-evolution -swift-version 5 \
+// RUN:   -enable-library-evolution \
+// RUN:   -target %target-swift-abi-5.8-triple \
 // RUN:   -emit-module %t/IndirectMixedDependency.swift \
 // RUN:   -emit-module-path %t/IndirectMixedDependency.swiftmodule
 // RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) \
-// RUN:   -enable-library-evolution -swift-version 5 \
+// RUN:   -enable-library-evolution \
+// RUN:   -enable-library-evolution \
```
Comment (tshortli) [inline test/TBD/multimodule-implied-objc.swift:15]: Minor: duplicate `-enable-library-evolution` flag

**Review by tshortli — APPROVED** [review]

**Comment by cyndyishida** [comment]
@swift-ci please test

**Comment by cyndyishida** [comment]
@swift-ci please test Linux

**Pull request merged**